### PR TITLE
Initial commit of OdomRaw message, used to send raw odom info.

### PR DESCRIPTION
**Pull request by markwomack**
Defining the raw odometry message that will be published from the arduino microcontroller (in our case the teensy) to the wider ros system. I don't know how general we want to get here, so currently focused on what we would be using to generate the geometry_msgs::TransformStamped and nav_msgs::Odometry messages.

### CODE REVIEW

File: arduino_output/OdomRaw.msg [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Message to transport raw odometry information from
+# the Arduino microcontroller to the main ROS system.
+# It is expected that this raw data will be expanded
+# into the full odometry messages used by ROS.
+
+Header header
+float64 x
```
Comment (griswaldbrooks) [inline arduino_output/OdomRaw.msg]: Let's make these Pose and Twist messages.

**Review by griswaldbrooks — CHANGES_REQUESTED** [review]
As in:
https://github.com/pal-robotics/pal_msgs

There should be a `msg` folder, a CMakeLists.txt, and a package.xml.

Let's not call it `arduino_output`, let's call it `victoria_nav_msgs` as this repo will contain many packages. If they're not preprended with `victoria_*` then when it's included in a package.xml it would look like `<depend>arduino_output</depend>` which is not ideal.

File: arduino_output/OdomRaw.msg (hunk 1 above)
Comment (mwimble) [inline arduino_output/OdomRaw.msg]: y? Did you mean x?
Griswald or Mark, will one of you be adding in the package.xml and CMakeLists.txt to cause the messages to be generated?

Comment (griswaldbrooks) [reply arduino_output/OdomRaw.msg]: I pinged @markwomack on the package and cmakelists a few minutes ago

**Comment by mwimble** [comment]
I’ll wait until this stabilizes and demonstrates message generation before I move my stuff there.

File: arduino_output/OdomRaw.msg (hunk 1 above)
Comment (markwomack) [reply arduino_output/OdomRaw.msg]: Pose contains a Point and Quaternian. Twist has two Vector3's. It is unclear to me that we are saving any space by using those compared to 5 float64's.

Comment (markwomack) [reply arduino_output/OdomRaw.msg]: Yes to adding the CMakeLists.txt and package.xml.  That was just my ignorance.

Comment (griswaldbrooks) [reply arduino_output/OdomRaw.msg]: The savings comes from that `nav_msgs/Odometry` 
http://docs.ros.org/api/nav_msgs/html/msg/Odometry.html
Contains these guys:
http://docs.ros.org/api/geometry_msgs/html/msg/PoseWithCovariance.html
http://docs.ros.org/api/geometry_msgs/html/msg/TwistWithCovariance.html
Each containing 36 `float64` saving 72 `float64`'s from being sent.

**Comment by markwomack** [comment]
OK, something more like what we want. I have not tried to build yet.

**Comment by markwomack** [comment]
So, question on ros message serialization...even if we never set a value on a field, it still gets serialized/deserialized?  So, why not leave OdomRaw as the 5 or 6 float64's?  I can see a Point maybe, but the other types will just impose transfer of more unneeded bytes.  Unless I am missing something fundamental here.  These 5 float64's are all that are needed to do the two messages we are trying to avoid sending from the teensy.

**Comment by markwomack** [comment]
Sorry, moving this discussion to the email thread "Cusom messages?"

**Comment by griswaldbrooks** [comment]
Ok, you make a good point. But in that case I wouldn't call the message `OdomRaw`.
I'd call it `Odom2DRaw` something else that emphasizes the 2D nature.
Use http://docs.ros.org/jade/api/geometry_msgs/html/msg/Pose2D.html and make a custom `Twist2D` message.

**Comment by griswaldbrooks** [comment]
I closed and reopened it to trigger travis

**Comment by griswaldbrooks** [comment]
@markwomack copy this file (https://github.com/victoriarobotics/ros-travis-integration/blob/master/.travis.yml) into your PR, so I can see what travis says.

**Comment by griswaldbrooks** [comment]
Into the root of the repo.

**Comment by mwimble** [comment]
I checked in code quite a while back. Is there a way I can manually trigger a travis build? What normally triggers a build?

**Comment by griswaldbrooks** [comment]
@mwimble There is a button there that says **Restart Build** but builds are normally triggered on new pushes.

**Comment by griswaldbrooks** [comment]
@mwimble where are you trying to trigger a build? On victoria_navigation?

**Comment by mwimble** [comment]
Yes

**Comment by griswaldbrooks** [comment]
So, as I said. But if you're trying to do it on your own fork, you'll have to sign up with Travis CI (which is free) and turn on your repo

**Comment by markwomack** [comment]
OK. It builds in my local workspace.

**Comment by griswaldbrooks** [comment]
@markwomack can you push that travis yaml?

**Comment by markwomack** [comment]
I don't know what the "travis yaml" you are referring to.  The one in victoria_platform?

**Comment by griswaldbrooks** [comment]
@markwomack I'd made this comment before

> @markwomack copy this file (https://github.com/victoriarobotics/ros-travis-integration/blob/master/.travis.yml) into your PR, so I can see what travis says.

**Comment by markwomack** [comment]
OK. Added .travis.yml.

**Comment by markwomack** [comment]
Updated to raw version of .travis.yml to make sure no parse errors.  Hopefully that gets it.

**Comment by markwomack** [comment]
Woo hoo!  All checks pass!

**Comment by mwimble** [comment]
Woo hoo, indeed.

**Pull request merged**